### PR TITLE
adds scorer to AggregateRequest

### PR DESCRIPTION
**Pull request by rbs333**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Do tests and lints pass with this change?
- [x] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [x] Is the new or changed code fully tested?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Is there an example added to the examples folder (if applicable)?
- [ ] Was the change added to CHANGES file?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

This PR adds support for setting the scorer with an aggregate call and resolves issue https://github.com/redis/redis-py/issues/3408.

### CODE REVIEW

**Comment by dmaier-redislabs** [comment]
Hi Robert, 

we are going to review this PR next week when @vladvildanov is back.

Regards,
David

**Comment by tylerhutcherson** [comment]
> Hi Robert, 
> 
> 
> 
> we are going to review this PR next week when @vladvildanov is back.
> 
> 
> 
> Regards,
> 
> David

Thanks @dmaier-redislabs @vladvildanov let me or Robert know if you need anything else, happy to help! Cheers.

**Comment by vladvildanov** [comment]
@rbs333 Hi! The following tests are failing

**Comment by rbs333** [comment]
@vladvildanov pushed a fix for tests

File: redis/commands/search/aggregation.py [hunk 1]
```diff
@@ -300,6 +301,17 @@ def add_scores(self) -> "AggregateRequest":
         self._add_scores = True
         return self
 
+    def scorer(self, scorer: str) -> "AggregateRequest":
+        """
+        Use a different scoring function to evaluate document relevance.
+        Default is `TFIDF`.
+
+        :param scorer: The scoring function to use
+                       (e.g. `TFIDF.DOCNORM` or `BM25`)
```
Comment (vladvildanov) [inline redis/commands/search/aggregation.py:310]: It makes sense to create an enum to handle different functions, in this way user will be aware of all possible values

**Review by vladvildanov — CHANGES_REQUESTED** [review]

**Comment by vladvildanov** [comment]
@rbs333 Left a comment after detailed review

File: redis/commands/search/aggregation.py (hunk 1 above)
Comment (rbs333) [reply redis/commands/search/aggregation.py:320]: @vladvildanov added

File: redis/commands/search/aggregation.py [hunk 2]
```diff
@@ -1,8 +1,18 @@
+from enum import Enum
 from typing import List, Union
 
 FIELDNAME = object()
 
 
+class Scorers(Enum):
+    TFIDF = "TFIDF"
```
Comment (rbs333) [inline redis/commands/search/aggregation.py:8]: https://redis.io/docs/latest/develop/interact/search-and-query/advanced-concepts/scoring/

available according to docs

File: redis/commands/search/aggregation.py (hunk 1 above)
Comment (vladvildanov) [reply redis/commands/search/aggregation.py:320]: The signature should be changed as well, or it makes no sense to the enduser.

`def scorer(self, scorer: Scorers) -> "AggregateRequest"`

Comment (rbs333) [reply redis/commands/search/aggregation.py:320]: done

**Pull request merged**